### DISCUSSION TITLE
Conditional Types?

### DISCUSSION THREAD

**Discussion by bestickley** [trpc/trpc#2888] [Q&A] [answered]
I have a `user.get` query that accepts an object typed like `{ id: string, includeRoles: boolean }`. The query returns either `User | UserWithRoles | undefined`. How can I create my TRPC router so that if `includeRoles` is `true`, the inferred return type is `UserWithRoles | undefined` and if it's false the return type is `User | undefined`?

**Comment by sachinraja** [comment, marked as answer]
That's not possible at the moment, see #2150. You can instead have two separate queries.
